src/pages/PostDetails/PostDetails.jsx

```diff
@@ -1,26 +1,26 @@
 import { LikeOutlined } from '@ant-design/icons'
 import { IconText, Loading } from 'components'
-import { useEffect, useState } from 'react'
+import { useQuery } from 'react-query'
 import { useParams } from 'react-router-dom'
+import { postService } from 'services'
 
 const PostDetails = () => {
   const { postId } = useParams()
-  const [post, setPost] = useState()
-  const [loading, setLoading] = useState(true)
 
-  useEffect(() => {
-    fetch(`http://localhost:3000/posts/${postId}`)
-      .then((response) => response.json())
-      .then((data) => {
-        setPost(data)
-      })
-      .finally(() => setLoading(false))
-  }, [])
+  const {
+    isLoading,
+    data: post,
+    error,
+  } = useQuery('post', () => postService.fetchPost(postId))
+
+  if (error) {
+    return <h1>{error.message}</h1>
```

Comment:
Please use also  navigation to `NotFound` page, when not found error is returned